rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -499,6 +512,50 @@ bool SequentialWriter::should_split_bagfile(
   return should_split;
 }
 
+void SequentialWriter::delete_oldest_files_if_needed()
+{
+  // Only delete if split count limit is set
+  // Note: This is only called after split_bagfile(), so max_bagfile_size is guaranteed to be set
+  if (storage_options_.max_splits == 0) {
+    return;
+  }
+
+  // Protect metadata access with mutex to prevent race conditions
+  std::lock_guard<std::mutex> lock(topics_info_mutex_);
+
+  // Delete oldest files until we're under the split-count limit
+  // Note: We just split, so we have at least 2 files and the oldest is definitely not current
+  while (metadata_.files.size() > 1) {
+    // If we're under the limit, stop deleting
+    if (metadata_.files.size() <= storage_options_.max_splits) {
+      break;
+    }
```

Comment:
```suggestion
   // Delete the oldest files until we're under the split-count limit
  while (metadata_.files.size() <= storage_options_.max_splits) {
```
Simplify logic.
Note: if `storage_options_.max_splits` were `storage_options_.max_bag_files`, the logic would be easy to understand.

Reply:
Good catch! Simplified.